src/main/java/tech/stackable/hadoop/StackableGroupMapper.java

```diff
@@ -0,0 +1,142 @@
+            this.opaUri = URI.create(opaMappingUrl);
+        } catch (Exception e) {
+            throw new OpaException.UriInvalid(opaUri, e);
+        }
+
+        this.mappingGroupName = configuration.get(OPA_MAPPING_GROUP_NAME_PROP);
+        if (mappingGroupName == null) {
+            throw new RuntimeException("Config \"" + OPA_MAPPING_GROUP_NAME_PROP + "\" missing");
+        }
+
+        this.json = new ObjectMapper()
+                // https://github.com/stackabletech/trino-opa-authorizer/issues/24
+                // OPA server can send other fields, such as `decision_id`` when enabling decision logs
+                // We could add all the fields we *currently* know, but it's more future-proof to ignore any unknown fields.
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+                // do not include null values
+                .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    }
+
+    /**
+     * Returns list of groups for a user.
+     *
+     * @param user get groups for this user
+     * @return list of groups for a given user
+     */
+    @Override
+    public List<String> getGroups(String user) throws IOException {
+        LOG.info("Calling StackableGroupMapper.getGroups for user [{}]", user);
+
+        HttpResponse<String> response = null;
+        OpaQuery query = new OpaQuery(new OpaQuery.OpaQueryInput(user));
+        String body = json.writeValueAsString(query);
+
+        LOG.debug("Request body [{}]", body);
+        try {
+            response = httpClient.send(
+                    HttpRequest.newBuilder(opaUri).header("Content-Type", "application/json")
+                            .POST(HttpRequest.BodyPublishers.ofString(body)).build(),
+                    HttpResponse.BodyHandlers.ofString());
+            LOG.info("Opa response [{}]", response.body());
+        } catch (InterruptedException e) {
+            LOG.error(e.getMessage());
+        }
+
+        switch (Objects.requireNonNull(response).statusCode()) {
+            case 200:
+                break;
+            case 404:
+                throw new OpaException.EndPointNotFound(opaUri.toString());
+            default:
+                throw new OpaException.OpaServerError(query.toString(), response);
+        }
+
+        String responseBody = response.body();
+        LOG.debug("Response body [{}]", responseBody);
+        List<String> groups = Lists.newArrayList();
+
+        @SuppressWarnings("unchecked")
+        Map<String, Object> result = (Map<String, Object>) json.readValue(responseBody, HashMap.class).get(OPA_RESULT_FIELD);
+        List<String> rawGroups = (List<String>) result.get(this.mappingGroupName);
+
+        for (String rawGroup :  rawGroups) {
+            groups.add(removeSlashes.apply(rawGroup));
+        }
+
+        LOG.info("Groups for [{}]: [{}]", user, groups);
+
+        return groups;
+    }
+
+    private final static UnaryOperator<String> removeSlashes = s -> {
```

Comment:
Puh, let's rewrite in [Rust](https://doc.rust-lang.org/std/primitive.str.html#method.trim_matches) xD

I think "strip" is a bit more expressive than "remove".

```suggestion
    private final static UnaryOperator<String> stripSlashes = s -> {
```